### PR TITLE
added possibility to kill CountContext

### PR DESCRIPTION
**Pull request by bogensberger**

### CODE REVIEW

File: sql/src/test/java/io/crate/jobs/CountContextTest.java [hunk 1]
```diff
@@ -62,4 +66,40 @@ public void testClose() throws Exception {
...
+    public void testKill() throws Exception {
+        ListenableFuture<Long> future = mock(ListenableFuture.class);
+        CountOperation countOperation = new FakeCountOperation(future);
+
+        RowDownstream rowDownstream = mock(RowDownstream.class);
+        when(rowDownstream.registerUpstream(any(RowUpstream.class))).thenReturn(mock(RowDownstreamHandle.class));
+        CountContext countContext = new CountContext(countOperation, rowDownstream, null, WhereClause.MATCH_ALL);
+
+        ContextCallback callback = mock(ContextCallback.class);
+        countContext.addCallback(callback);
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/jobs/CountContextTest.java]: is the callback required?

File: sql/src/test/java/io/crate/jobs/CountContextTest.java [hunk 2]
```diff
@@ -62,4 +66,40 @@ public void testClose() throws Exception {
         future.setException(new UnknownUpstreamFailure());
         verify(callback, times(1)).onClose();
     }
+
+    @Test
+    public void testKill() throws Exception {
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/jobs/CountContextTest.java]: rename to something more specific... testKillCancelsOperationFuture or something like that

File: sql/src/test/java/io/crate/jobs/CountContextTest.java (hunk 1 above)
Comment (bogensberger) [inline sql/src/test/java/io/crate/jobs/CountContextTest.java]: sry, forgot to verify that the callback is closed which i also want to test

**Comment by mfussenegger** [comment]
ack

**Pull request merged**